app/src/main/java/com/khuttun/notificationnotes/NotificationMgr.java

```diff
@@ -90,43 +85,158 @@ public void setGroupNotification(ArrayList<NotificationNote> notes)
                 break;
 
             default:
-                this.notificationManager.notify(GROUP_NOTIF_ID, createGroupNotification(visibleNotes));
+                if (groupAPI)
+                {
+                    ArrayList<Notification> groupedNotes = createGroupNotifications(visibleNotes);
+                    for (int i = 0; i < groupedNotes.size(); i++)
+                    {
+                        Notification note = groupedNotes.get(i);
+                        String title = note.extras.getString("android.title");
+                        String text = note.extras.getString("android.text");
+                        int id = -1;
+                        if (i == groupedNotes.size() - 1)
+                        {
+                            id = SUMMARY_NOTIF_ID;
+                        }
+                        else if (i == 0)
+                        {
+                            id = GROUP_NOTIF_ID;
+                        }
+                        else
+                        {
+                            id = findNotificationID(notes, title, text);
+                        }
+                        this.notificationManager.notify("note", id, note);
+                    }
+                }
+                else
+                {
+                    this.notificationManager.notify("note", SUMMARY_NOTIF_ID,
+                            createSummaryNotification(visibleNotes));
+                }
                 break;
         }
     }
 
+    private int findNotificationID(ArrayList<NotificationNote> notes, String title, String text)
```

Comment:
This function doesn't really work right if you happen to have two notes with identical title and text. Would it be possible to restructure code so that function like this wouldn't be needed?